src/utils/charsets.cpp

```diff
@@ -0,0 +1,195 @@
+/*
+ *  Copyright (C) 2024 Team Kodi (https://kodi.tv)
+ *  Copyright (C) 2015 Jan van Katwijk (J.vanKatwijk@gmail.com)
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSE.md for more information.
+ */
+
+/*
+  * This file is taken from:
+  * https://github.com/AlbrechtL/welle.io/blob/v2.4/src/backend/charsets.cpp
+  * Text below taken from original and nothing todo with coding here.
+  * License: GPL-2.0-or-later
+  * Copyright (C): 2015 Jan van Katwijk (J.vanKatwijk@gmail.com)
+  * Name: Lazy Chair Programming
+  * Note: This charset handling was kindly added by Przemyslaw Wegrzyn all rights acknowledged
+  */
```

Comment:
Hi @ksooo, can you look here? Not sure how the best style to cleanup and match our style and still refer code good as possible to original. Maybe a better idea?